From 09fe887fd1b607ed4a649b84fc41a740c70aab01 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Tue, 19 May 2015 17:50:55 +0800
Subject: [PATCH] HDMI: fix edid 3d information offset parse error.

	In EDID 3D descriptor, 3d information is after
	4K format. But we did not increase the offset of
	3d content if 4K format is zero, which make 3d
	information error.

Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/video/rockchip/hdmi/rockchip-hdmi-edid.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmi-edid.c b/drivers/video/rockchip/hdmi/rockchip-hdmi-edid.c
index f4541edffebd..48597f1bcdaf 100644
--- a/drivers/video/rockchip/hdmi/rockchip-hdmi-edid.c
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmi-edid.c
@@ -181,7 +181,8 @@ static int hdmi_edid_parse_3dinfo(unsigned char *buf, struct list_head *head)
 		len = (buf[1] & 0xe0) >> 5;
 		for (i = 0; i < len; i++) {
 			if (buf[offset])
-				hdmi_add_vic((96 - buf[offset++]), head);
+				hdmi_add_vic((96 - buf[offset]), head);
+			offset++;
 		}
 	}
 
-- 
2.35.3

